src/components/PriceBar/TileColorBoard.tsx

```diff
@@ -1,9 +1,56 @@
 import { Center, Flex, Text } from "@chakra-ui/react";
-import React from "react";
+import React, { useEffect } from "react";
 import { useStoreSelector } from "@/store/hooks";
+import { IPriceCalculation, ITilePrice, ICourts } from "../../interfaces/priceCalculation";
+import { useGetPriceQuery } from "@/redux/api/priceApi";
+import { useDispatch } from "react-redux";
+import { changeTotalPrice } from "@/store/reducer/totalPriceSlice";
+import priceFormat from "@/utils/priceFormat";
 
 const TileColorBoard: React.FC = () => {
   const tileBlocks = useStoreSelector((state) => state.priceBar.blocks);
+  const dispatch = useDispatch();
+  const courts = useStoreSelector((state) => state.courtSize);
+  const budget = useStoreSelector((state) => state.totalPrice.budget);
+  const { data } = useGetPriceQuery(0);
+
+  useEffect(() => {
+    let tilePrice = 0;
+    let installPrice = 0;
+    let deliveryPrice = 0;
+    let totalQuantity = 0;
+    if (data) {
```

Comment:
You can remove if (data) by simply using question mark, same for the rest.   
const priceList = data?.find((item: IPriceCalculation) => !item.isDeleted);